### PR TITLE
Edit component groups. Closes #452.

### LINKED ISSUE #452: Edit component groups
We have no facility to edit component groups.

### PR DESCRIPTION
**Pull request by jbrooksuk**

### CODE REVIEW

**Comment by joecohens** [comment]
:+1:

File: src/Http/Controllers/DashComponentController.php [hunk 1]
```diff
@@ -303,4 +318,45 @@ public function postAddComponentGroup()
...
+            return Redirect::back()->withInput(Binput::all())
+                ->with('title', sprintf(
+                    '<strong>%s</strong> %s',
+                    trans('dashboard.notifications.whoops'),
+                    trans('dashboard.components.groups.edit.failure')
+                ))
+                ->with('errors', $group->getErrors());
+        }
+
+        segment_track('Dashboard', [
```
Comment (joecohens) [inline src/Http/Controllers/DashComponentController.php]: @jbrooksuk What do you think if we fire events to log this stuff?

Comment (jbrooksuk) [inline src/Http/Controllers/DashComponentController.php]: Such as?

Comment (jbrooksuk) [inline src/Http/Controllers/DashComponentController.php]: Oh, you mean using Laravel events and moving the Segment code into them?

Comment (joecohens) [inline src/Http/Controllers/DashComponentController.php]: Lol yeah :)

Comment (jbrooksuk) [inline src/Http/Controllers/DashComponentController.php]: Yes, we should definitely do that.

**Pull request merged**